src/RowsInput/index.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { Button, ButtonGroup } from '..';
+
+export interface IRowInputProps<T> {
+  key: string;
+  row: T;
+  onRemove: () => void;
+  onChange: (row: T) => void;
+}
+
+export interface IRowsInputProps<T> {
+  rows: T[];
+  text: string;
+  onAddRow: () => void;
+  onRemoveRow: (i: number) => void;
+  onChangeRow: (i: number, row: T) => void;
+  RowInput: React.FunctionComponent<IRowInputProps<T>>;
+}
+
+export function RowsInput<T>({ rows, text, onAddRow, onRemoveRow, onChangeRow, RowInput }: IRowsInputProps<T>) {
+  const [rowsShiftedIndex, shiftRows] = React.useReducer(i => i + 1, 0);
+  return (
+    <>
+      <ButtonGroup>
+        <Button icon="plus" text={text} onClick={onAddRow} />
+      </ButtonGroup>
```

Comment:
Instead of this add button, I suggest adding an empty row at the end of the list that calls `onAddRow` onFocus. You can see how we're doing this in the RequestMaker component for the parameters list: https://github.com/stoplightio/elements/blob/master/src/components/RequestMaker/Request/Parameters.tsx#L142-L148